### PR TITLE
Enhanced RDoc for Enumerable#inject

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Earlier effort was unsatisfactory.  Trying again.

### CODE REVIEW

File: enum.c [hunk 1]
```diff
@@ -871,44 +871,138 @@ ary_inject_op(VALUE ary, VALUE init, VALUE op)
...
+ *  - A block to which each operand is passed.
+ *
+ *  With method-name argument +symbol+,
+ *  combines operands using the method:
+ *
+ *    # Sum, without initial_operand.
+ *    (1..4).inject(:+)     # => 10
+ *    # Sum, with initial_operand.
+ *    (1..4).inject(10, :+) # => 20
+
```
Comment (jeremyevans) [inline enum.c:891]: Empty line should be removed.

File: enum.c [hunk 2]
```diff
@@ -871,44 +871,138 @@ ary_inject_op(VALUE ary, VALUE init, VALUE op)
...
+ *    # Integer multiplication.
+ *    (1..4).inject(:*)                # => 24
+ *    # Character range concatenation.
+ *    ('a'..'d').inject('', :+)        # => "abcd"
+ *    # String array concatenation.
+ *    %w[foo bar baz].inject('', :+)   # => "foobarbaz"
+ *    # Hash update.
+ *    h = [{foo: 0, bar: 1}, {baz: 2}, {bat: 3}].inject(:update)
+ *    h # => {:foo=>0, :bar=>1, :baz=>2, :bat=>3}
+ *    # Hash appending.
```
Comment (JuPlutonic) [inline enum.c:973]: # Hash to array conversion.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good, can be committed without further review after the empty line is removed.

File: enum.c (hunk 2 above)
Comment (BurdetteLamar) [reply enum.c]: > 
> 
> `# Hash to array conversion.`

# Hash conversion to nested arrays.

Done.

**Pull request merged**